include/MappingSettings.hpp

```diff
@@ -1,95 +1,182 @@
+            return "commander";
+        case Encodings::Bimander:
+            return "bimander";
+    }
+    return " ";
+}
+
+NLOHMANN_JSON_SERIALIZE_ENUM(Encodings, {
+                                                {Encodings::None, "none"},
+                                                {Encodings::Commander, "commander"},
+                                                {Encodings::Bimander, "bimander"},
+                                        })
+
+static std::string toString(const Groupings grouping) {
+    switch (grouping) {
+        case Groupings::Fixed2:
+            return "fixed2";
+        case Groupings::Fixed3:
+            return "fixed3";
+        case Groupings::Logarithm:
+            return "logarithm";
+        case Groupings::Halves:
+            return "halves";
+    }
+    return " ";
+}
+
+NLOHMANN_JSON_SERIALIZE_ENUM(Groupings, {
+                                                {Groupings::Fixed2, "fixed2"},
+                                                {Groupings::Fixed3, "fixed3"},
+                                                {Groupings::Halves, "halves"},
+                                                {Groupings::Logarithm, "logarithm"},
+                                        })
+
+static std::string toString(const Strategy strategy) {
+    switch (strategy) {
+        case Strategy::ArchitectureSwaps:
+            return "architectureswaps";
+        case Strategy::SubsetSwaps:
+            return "subsetswaps";
+        case Strategy::Custom:
+            return "custom";
+        case Strategy::None:
+            return "none";
+        case Strategy::Increasing:
+            return "increasing";
+    }
+    return " ";
+}
+
+NLOHMANN_JSON_SERIALIZE_ENUM(Strategy, {{Strategy::None, "none"},
+                                        {Strategy::ArchitectureSwaps, "architectureswaps"},
+                                        {Strategy::SubsetSwaps, "subsetswaps"},
+                                        {Strategy::Custom, "custom"},
+                                        {Strategy::Increasing, "increasing"}})
+
+struct MappingSettings {
+    MappingSettings() = default;
+
+    unsigned int timeout = 3600000; // 60min timeout
+    void         setTimeout(unsigned int sec) { timeout = sec; }
+
+    LayeringStrategy layeringStrategy = LayeringStrategy::None;
+
+    /// Settings for heuristic approach
+    InitialLayoutStrategy initialLayoutStrategy = InitialLayoutStrategy::None;
+
+    bool admissibleHeuristic = true;
+    bool verbose             = false;
+
+    bool               lookahead            = true;
+    int                nrLookaheads         = 15;
+    int                teleportationQubits  = 0;
+    unsigned long long teleportationSeed    = 0;
+    bool               teleportationFake    = false;
+    double             firstLookaheadFactor = 0.75;
+    double             lookaheadFactor      = 0.5;
+    Encodings          encoding             = Encodings::None;

```

Comment:
not in general, while the encodings are faster, esp. for larger circuits, smaller circuits actually are faster without the overhead

Reply:
Is there some sort of "threshold" for when it makes sense to switch to the encodings?
Is it small circuits with respect to the number of qubits or rather small circuits regarding the gate count?
It would be great to have some sensible defaults that are chosen automatically (maybe even depending on the benchmark)

Reply:
both actually, but the qubit count is the more important.
I don't really like when programs choose options for me automatically, other than a given default that works in the majority of cases, so that is something I always try to avoid (because when I'm the user, I then often wonder why suddenly the config changes and have to dig through the documentation anyway)